### PR TITLE
chore: add license header to files and github action workflow to check license

### LINKED ISSUE #149: Check the license header for notation-core-go and its dependencies
### Is your feature request related to a problem?

Consider if notation-core-go needs dependency license checking and reporting

### What solution do you propose?

Consider adding dependency license checking and reporting similar to https://github.com/notaryproject/notation/issues/706

### What alternatives have you considered?

If notation-core-go does not require/need this, clearly document in README or other public location

### Any additional context?

_No response_

### PR DESCRIPTION
**Pull request by ghost**
Resolves #149

### CODE REVIEW

File: .github/workflows/license-checker.yml
Comment (shizhMSFT) [inline .github/workflows/license-checker.yml:1]: Can we make this workflow reusable like `build` and `codeql` for `notation-go`, `notation-core-go`, and potentially `notation`?

**Review by shizhMSFT — DISMISSED** [review]
LGTM with suggestions

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: .github/workflows/license-checker.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: License Checker
+
+on:
+  push:
+    branches:
+      - main
+      - release-*
```
Comment (shizhMSFT) [inline .github/workflows/license-checker.yml:20]: We don't have release branch for libraries.

File: .github/workflows/license-checker.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  pull_request:
+    branches:
+      - main
+      - release-*
```
Comment (shizhMSFT) [inline .github/workflows/license-checker.yml:24]: We don't have release branch for libraries.

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Pull request merged**